python/ray/data/impl/arrow_block.py

```diff
@@ -377,7 +377,9 @@ def _copy_table(table: "pyarrow.Table") -> "pyarrow.Table":
     cols = table.columns
     new_cols = []
     for col in cols:
-        if col.num_chunks > 0 and isinstance(col.chunk(0), pa.ExtensionArray):
+        if col.num_chunks == 0:
+            arr = pa.chunked_array([], type=col.type)
```

Comment:
I'm curious about this change, was it that `col.combine_chunks()` doesn't properly preserve the array type if there are no chunks?

Also, we need to have that `arr` is a `pa.Array`, not a `pa.ChunkedArray`, after this if-else statement, so this change would break that assumption.

Reply:
I remember combine_chunks will fail if there are no chunks.